internal/cmd/cmd.go

```diff
@@ -200,17 +199,22 @@ $ infra use kubernetes.development.kube-system
 				return err
 			}
 
-			parts := strings.Split(name, ".")
+			parts := strings.Split(destination, ".")
```

Comment:
Seems we rely on there being at most 2 parts, so maybe this?

```suggestion
			parts := strings.SplitN(destination, ".", 2)
```

Reply:
Ah it's possible the user supplies more than 2 parts, e.g. `kubernetes.cluster.namespace`, in which case we want to discard the first part and pass the second and third part to `SetContext`